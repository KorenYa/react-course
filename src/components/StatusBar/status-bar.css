.status-bar {
    padding: 10px 40px;
    background-color: #f5f5f5;
    font-weight: 600;
}

.status-bar__filters {
    float: right;
}

.status-bar__sort {
    cursor: pointer;
    padding: 0 10px;
}

.status-bar__sort.active {
    color: #f55263;
}