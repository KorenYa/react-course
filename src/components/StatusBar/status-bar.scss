@import '../../styles/constants.scss';

.status-bar {
    padding: 10px 40px;
    background-color: $light-grey;
    font-weight: $bold;

    &__filters {
        float: right;
    }

    &__sort {
        cursor: pointer;
        padding: 0 10px;

        &.active {
            color: $red;
        }
    }
}
