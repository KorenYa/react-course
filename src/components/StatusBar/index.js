import React from 'react';
import PropTypes from 'prop-types';
import './status-bar.scss';
const StatusBar = function(props) {
    const noHits = <span>&nbsp;</span>;

    const status = (
        <div>
            {props.total} movies found
            <div className='status-bar__filters'>
                <span>Sort by</span>
                <span className='status-bar__sort'>release date</span>
                <span className='status-bar__sort'>rating</span>
            </div>
        </div>
    );

    return (
        <div className='status-bar'>{props.total > 0 ? status : noHits}</div>
    );
};
StatusBar.propTypes = {
    total: PropTypes.number
};
export default StatusBar;
