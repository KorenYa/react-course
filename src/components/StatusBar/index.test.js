import React from 'react';
import renderer from 'react-test-renderer';

import StatusBar from '../StatusBar';

describe('StatusBar', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<StatusBar total={3} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
