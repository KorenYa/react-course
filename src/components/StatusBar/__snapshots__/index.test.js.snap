// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['StatusBar renders correctly 1'] = `
<div
  className="status-bar"
>
  <div>
    3
     movies found
    <div
      className="status-bar__filters"
    >
      <span>
        Sort by
      </span>
      <span
        className="status-bar__sort active"
        onClick={[Function]}
      >
        release date
      </span>
      <span
        className="status-bar__sort "
        onClick={[Function]}
      >
        rating
      </span>
    </div>
  </div>
</div>
`;
